pilot/pkg/bootstrap/server.go

```diff
@@ -147,22 +143,29 @@ func NewServer(args PilotArgs) (*Server, error) {
 		}
 	}
 
-	s := &Server{
-		fileWatcher: filewatcher.NewWatcher(),
-	}
-
-	prometheus.EnableHandlingTimeHistogram()
-
-	// Apply the arguments to the configuration.
-	if err := s.initKubeClient(&args); err != nil {
+	kubeClient, err := newKubeClient(&args)
```

Comment:
Not sure if that was the code before - but we need to stop creating kube clients in every module, just reuse the existing one.